components/serialization/json/spotBugsExcludeFilter.xml

```diff
@@ -26,4 +26,8 @@ xsi:schemaLocation="https://github.com/spotbugs/filter/3.0.0 https://raw.githubu
 			<Class name="com.microsoft.kiota.serialization.UnionWrapperParseTests$4$1" />
 		</Or>
 	</Match>
+	<Match>
+		<Bug pattern="NP_BOOLEAN_RETURN_NULL" />
```

Comment:
should we fix this issue instead of excluding it?

Reply:
I believe this is an intended implementation, though I think a comment should be added to explain the reasoning. 
``` Java
    @Nullable
    public Boolean getBooleanValue() {
        return currentNode.isJsonPrimitive() ? currentNode.getAsBoolean() : null;
    }
```
Always returning a boolean in this above method would suggest that the parseNode has a valid boolean value, which, if the node is not a jsonPrimitive would not be true. By retuning 'null' we avoid throwing an exception and rather return 'null' to suggest that it does not have a valid boolean value. The Nullable annotation is there to document that this method may return a null value, however SpotBugs continues to flag this method. My opinion is that a comment should be added to the getBooleanValue method in the ParseNode.java interface within the abstraction library to communicate the meaning of a null value.